### PR TITLE
some refactorings

### LINKED ISSUE #16: get some error since the opam-repository windows support landed
since https://github.com/ocaml/opam-repository/pull/25861 got merged, I can see the following error, which should be looked into further:

`6m40s [ORB] error in duni_deps: set_of_formula: couldn't figure out or (((system-mingw) | system-msvc)) | (((system-mingw) | system-msvc))`

### PR DESCRIPTION
**Pull request by hannesm**
this also fixes #16

### CODE REVIEW

File: src/duni_deps.ml [hunk 1]
```diff
@@ -8,14 +8,19 @@
...
+    | "monorepo" | "vendor" | "post" -> Some (OpamTypes.B true)
+    | "build" | "with-test" | "dev" | "with-doc" -> Some (B false)
+    | "os" -> Option.map (fun x -> OpamTypes.S x) vars.Common.os
+    | "os-family" -> Option.map (fun x -> OpamTypes.S x) vars.Common.os_family
+    | "os-distribution" -> Option.map (fun x -> OpamTypes.S x) vars.Common.os_distribution
+    | "os-version" -> Option.map (fun x -> OpamTypes.S x) vars.Common.os_version
+    | "arch" -> Option.map (fun x -> OpamTypes.S x) vars.Common.arch
     | _ -> None
   in
-  OpamFilter.filter_formula ~default:false env depends
```
Comment (hannesm) [inline src/duni_deps.ml:18]: the reason to remove this `~default:false` is that we notice future missing variables. we can re-add it if it is annoying.

the code is called from Orb with a `try .. catch` anyways, thus the `filter_formula` raising an exception is handled.

File: src/orb.ml [hunk 2]
```diff
@@ -77,20 +77,29 @@ let custom_env_keys = [
...
+  os_family = OpamSysPoll.os_family gt_vars ;
+  arch = OpamSysPoll.arch gt_vars ;
+}
+
+let custom_env vars s = [
+  "OS", vars.Common.os;
+  "OS_DISTRIBUTION", vars.os_distribution;
+  "OS_VERSION", vars.os_version;
+  "OS_FAMILY", vars.os_family;
+  "ARCH", vars.arch;
```
Comment (hannesm) [inline src/orb.ml:93]: we now include arch as well.

File: src/orb.ml [hunk 3]
```diff
@@ -104,51 +113,32 @@ let env_of_string str =
...
-      | Some "freebsd" ->
-        let r = Sys.command ("/usr/sbin/pkg query %n-%v > " ^ filename) in
-        if r <> 0 then log "failed to dump system packages (exit %d)" r
-      | Some distrib ->
-        log "unsupported OS for host system packages (bsd %s)" distrib
-      | None ->
-        log "unsupported OS (no system packages), bsd without distribution"
-    end
-  | Some "debian" ->
+let dump_system_packages ~os ~os_family filename =
```
Comment (hannesm) [inline src/orb.ml:116]: as mentioned in the commit message, using os and os_family looks sufficient (for now).

**Review by reynir — APPROVED** [review]
This looks nice to me

**Pull request merged**